### PR TITLE
Cache result of `nativeLink`

### PR DESCRIPTION
**Pull request by Duhemm**
So that it won't be run again if no input file has changed and if none
of the relevant scala-native options has changed.

### CODE REVIEW

File: sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala [hunk 1]
```diff
@@ -178,6 +200,9 @@ object ScalaNativePluginInternal {
     }
   )
 
+  private def writeConfig(file: File, config: Any*): Unit =
```
Comment (densh) [inline sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: `writeConfigHash`

File: sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala [hunk 2]
```diff
@@ -178,6 +200,9 @@ object ScalaNativePluginInternal {
...
+    IO.write(file, config.##.toString)
```
Comment (densh) [inline sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Something like sbt's `sbt.Hash(configString)` would be more robust. Scala's hash is only 32 bits, compared to 160 bits of SHA-1 of `sbt.Hash`.

File: sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala (hunk 1 above)
Comment (Duhemm) [inline sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Done

File: sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala (hunk 2 above)
Comment (Duhemm) [inline sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala]: Done

**Comment by densh** [comment]
LGTM!

**Pull request merged**